libr/util/p_format.c

```diff
@@ -996,10 +996,25 @@ static void r_print_format_nulltermstring(const RPrint* p, const int len, int en
 		}
 	} else if (MUSTSEEJSON) {
 		int j = i;
+
 		p->cb_printf ("%d,\"string\":\"", seeki);
 		for (; j < len && ((size == -1 || size-- > 0) && buf[j]) ; j++) {
 			if (IS_PRINTABLE (buf[j])) {
-				p->cb_printf ("%c", buf[j]);
+				/* encoding here so the format of . for non printable chars will stay */
+				int k = 0;
+				size_t encoded_length = 0;
+				char *encoded_char = r_str_utf16_encode ((const char *)(buf + j), 1);
+
+				if (encoded_char){
```

Comment:
missing space after )

Reply:
why not just encode the whole string and print it at once? instead of looping char by char?

Reply:
no reason, thanks